README.md

```diff
@@ -32,25 +32,82 @@ Please have patience as we update those to `awsm`.
 - *Simplicity.*
 	The entire code base for the compiler and runtime is relatively small.
 	The compiler is <3.5K lines of Rust, and the runtime (for *all* platforms) is <5K lines of C.
 	It is nearly trivial to implement different means of sandboxing memory accesses.
 	We've implemented *seven* different mechanisms for this!
 - *Portability.*
 	Both the compiler and runtime are mostly platform-independent code, and porting to a new platform only really requires additional work if you need to tweak stack sizes (microcontrollers), or use architectural features (e.g., MPX, segmentation, etc...).
+- *Composability.*
+	The final output of aWsm is simple `*.o` elf objects that can be linked into larger systems.
+	This enables the trivial composition of sandboxes together, and sandboxes into larger programs.
 
-We believe that aWsm is one of the best options for ahead-of-time compilation for outside of the browser.
+We believe that aWsm is one of the best options for ahead-of-time compilation for Wasm execution outside of the browser.
+
+If you want to learn more about aWsm, see the [design](doc/design.md), or the [publication](https://www2.seas.gwu.edu/~gparmer/publications/emsoft20wasm.pdf).
 
 # Performance
 
-Give us a little time, we'll post benchmarks here!
+PolyBench/C benchmarks for **x86-64** (slowdown over native C):
+
+| | Wasmer | WAVM | Node.js + Emscripten | Lucent | aWsm |
+| --- | --- | --- | --- | --- | --- |
+| Avg. Slowdown | 149.8% | 28.1% | 84.0% | 92.8% | 13.4% |
+| Stdev. in Slowdown | 194.09 | 53.09 | 107.84 | 117.25 | 34.65 |
+
+PolyBench/C benchmarks for **Arm Aarch64** (slowdown over native C):
+
+| | aWsm |
+| --- | --- |
+| Avg. Slowdown | 6.7% |
+| Stdev. of Slowdown | 19.38 |
+
+Polybench/C benchmarks for **Arm Cortex-M** microcontrollers (slowdown over native C):
+
+| Architectures | aWsm |
+| --- | --- |
+| Cortex-M7 Avg. slowdown | 40.2% |
+| Cortex-M4 Avg. slowdown | 24.9% |
+
+In comparison, the `wasm3` interpreter's slowdown on microcontrollers is more than 10x.
+For more details (including other bounds checking mechanisms), see the [paper](https://www2.seas.gwu.edu/~gparmer/publications/emsoft20wasm.pdf).
+
+There are many compelling runtimes, but we believe that aWsm is useful in generating very fast code, while being simple and extensible.
+
+## Comparison to Existing Wasm Ecosystems
+
+There are many existing compilers and runtimes.
+aWsm fills the niche of a compiler
+
+- based on ahead-of-time compilation using the popular LLVM infrastructure,
+- that generates fast, safe code (even on microcontrollers), and
+- that is designed for simplicity and easy extension.
+
+Adding runtime functions and changing safety checking mechanisms are trivial operations.
+
+We annotate if - to the best of our knowledge - each system uses *Interpretation (Int)*, *Just in Time (JIT)*, or *Ahead-of-Time (AoT)*, and if the infrastructure supports *Arm* (Cortex-A processors) and *Microcontrollers (uCs)* (all support x86).
```

Comment:
I'm thinking `uC` is too generic, applies to a wide range of microcontrollers outside of Arm also. Should this be more specific? Not sure. 

Another point is, x86 vs x86_64? I think, we mean both x86 and x86_64? It could be confusing.

Reply:
WAVM only supports x86_64 doesn't it?

Hmm, maybe a table would work better here?

Another concern is are we going to be able to maintain this? The WebAssembly space moves fast....

Runtime | Method | x86_64 | x86 | aarch64 | thumb | URL
---------|----------|--------|------|---------|--------|----
aWsm | AoT | ✓ | ✓ | ✓ | ✓ | You are here
Wasmtime | AoT | ✓ | ✓ | ✓ |  | https://github.com/bytecodealliance/wasmtime
Wasmer | AoT | ✓ | ✓ | ✓ |  | https://github.com/wasmerio/wasmer
WAMR | Pseudo-AoT/Int | ✓ | ✓ | ✓ | ✓ | https://github.com/bytecodealliance/wasm-micro-runtime
Wasm3 | Int | ✓ | ✓ | ✓ | ✓ | https://github.com/wasm3/wasm3
Wasmi | Int | ✓ | ✓ | ✓ | ✓ | https://github.com/paritytech/wasmi

Reply:
@phanikishoreg WAVM only supports x86_64 doesn't it?

Hmm, maybe a table would work better here?

Another concern is are we going to be able to maintain this? The WebAssembly space moves fast....

Runtime | Method | x86_64 | x86 | aarch64 | thumb | URL
---------|----------|--------|------|---------|--------|----
aWsm | AoT | ✓ | ✓ | ✓ | ✓ | You are here
Wasmtime | AoT | ✓ | ✓ | ✓ |  | https://github.com/bytecodealliance/wasmtime
Wasmer | AoT | ✓ | ✓ | ✓ |  | https://github.com/wasmerio/wasmer
WAMR | Pseudo-AoT/Int | ✓ | ✓ | ✓ | ✓ | https://github.com/bytecodealliance/wasm-micro-runtime
Wasm3 | Int | ✓ | ✓ | ✓ | ✓ | https://github.com/wasm3/wasm3
Wasmi | Int | ✓ | ✓ | ✓ | ✓ | https://github.com/paritytech/wasmi

Reply:
My bad, I was thinking about awsm when I wrote that x86 comment! Yes, I think, wavm only supports 64 bit coz of the virtual memory based sandboxing. 

I like the table, and you make a good point about maintaining the table. I think it might hold for the performance numbers too if you think about it. Perhaps we write the last updated date around the tables at the very least.

Reply:
github will not allow me to "edit" your table, thus copy the mardown. Any chance you post it again

```
quoted out
```
? @geky

Reply:
Nevermind, emacs-fu got me there ;-)